tests/components/onkyo/test_config_flow.py

```diff
@@ -0,0 +1,287 @@
+"""Test Onkyo config flow."""
+import asyncio
+
+from homeassistant import config_entries, core
+from homeassistant.components.onkyo.const import (
+    CONF_ENABLED_SOURCES,
+    CONF_IDENTIFIER,
+    CONF_MAX_VOLUME,
+    CONF_SOURCES,
+    DOMAIN,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from tests.common import AsyncMock, Mock, MockConfigEntry, patch
+
+TEST_NAME_1 = "OnkyoReceiver"
+TEST_NAME_2 = "PioneerReceiver"
+TEST_HOST_1 = "192.168.1.2"
+TEST_HOST_2 = "192.168.1.3"
+TEST_IDENTIFIER_1 = "0123456789"
+TEST_IDENTIFIER_2 = "9876543210"
+
+TEST_MAX_VOLUME = 200
+TEST_ENABLED_SOURCES = ["tv", "dvd"]
+TEST_CUSTOM_SOURCE_NAMES = {
+    "tv": "CustomNameTV",
+    "dvd": "CustomNameDVD",
+}
+
+
+def get_mock_connection(host, name, identifier):
```

Comment:
It is recommended that config_flow tests do not hit the integration internals.
You should look at adding `mock_setup_entry` in a local `conftest.py` file.
```suggestion
pytestmark = pytest.mark.usefixtures("mock_setup_entry")


def get_mock_connection(host, name, identifier):
```